test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -494,24 +498,26 @@ public final void testMeta() throws IOException {
         );
     }
 
-    public final void testDeprecatedBoost() throws IOException {
+    public final void testDeprecatedBoostWarning() throws IOException {
         try {
             createMapperService(DEPRECATED_BOOST_INDEX_VERSION, fieldMapping(b -> {
                 minimalMapping(b, DEPRECATED_BOOST_INDEX_VERSION);
                 b.field("boost", 2.0);
             }));
             String[] warnings = Strings.concatStringArrays(
-                getParseMinimalWarnings(),
+                getParseMinimalWarnings(DEPRECATED_BOOST_INDEX_VERSION),
                 new String[] { "Parameter [boost] on field [field] is deprecated and has no effect" }
             );
             assertWarnings(warnings);
         } catch (MapperParsingException e) {
             assertThat(e.getMessage(), anyOf(containsString("Unknown parameter [boost]"), containsString("[boost : 2.0]")));
         }
+    }
 
+    public void testBoostNotAllowed() throws IOException {
```

Comment:
I've separated boost checking in two, for deprecated warnings and boosts not allowed.